src/WalletModal.tsx

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import { EIP6963ProviderDetail } from './stores/walletStore';
+import styles from './WalletModal.module.css';
+import { ButtonIcon } from 'lib-kava-ai';
+import { X } from 'lucide-react';
+import metamaskLogo from './assets/MetaMask-icon-fox.svg';
+import hotWalletLogo from './assets/HOT Wallet Short.svg';
+
+interface WalletModalProps {
+  onClose: () => void;
+  availableProviders: EIP6963ProviderDetail[];
+  onSelectProvider: (provider: EIP6963ProviderDetail) => void;
+}
+
+const WalletModal: React.FC<WalletModalProps> = ({
+  onClose,
+  availableProviders,
+  onSelectProvider,
+}) => {
+  const hasProviders = availableProviders.length > 0;
+  return (
+    <div className={styles.providerModalBackdrop}>
+      <div className={styles.providerModal}>
+        <div className={styles.modalHeader}>
+          <h3>{hasProviders ? 'Select a Wallet' : 'Get a Wallet'}</h3>
+          <ButtonIcon
+            icon={X}
+            aria-label={'Close wallet connect'}
+            onClick={onClose}
+          />
+        </div>
+
+        <div className={styles.providersList}>
+          {hasProviders ? (
+            availableProviders.map((provider) => (
```

Comment:
If the user has EIP-6963 compatible wallet providers installed, they will be rendered as options to choose.

If they don't have any, then we provide a hard coded lists of options that they can install.